components/dashboard/src/workspaces/BrowserExtensionBanner.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useState } from "react";
+import UAParser from "ua-parser-js";
+import flashIcon from "../icons/Flash.svg";
+
+interface BrowserOption {
+    aliases?: string[];
+    url: string;
+}
+
+const installationOptions: Record<string, BrowserOption> = {
+    firefox: {
+        url: "https://addons.mozilla.org/en-US/firefox/addon/gitpod/",
+    },
+    chrome: {
+        aliases: ["edge", "brave", "chromium", "vivaldi", "opera"],
+        url: "https://chrome.google.com/webstore/detail/gitpod-always-ready-to-co/dodmmooeoklaejobgleioelladacbeki",
+    },
+};
+
+interface BrowserExtensionBannerProps {
+    parser?: UAParser;
+}
+
+export function BrowserExtensionBanner({ parser = new UAParser() }: BrowserExtensionBannerProps) {
```

Comment:
This will end up creating a new instance of `UAParser` on each render. Since we don't pass in a parser here in the one place we use it, it seems safe to just memoize the instance in this component instead of handling it via a prop. That way we only instantiate the parser once.

```ts
const parser = useMemo(() => new UAParser(), []);
```